### PR TITLE
Adds authentication mechanism and user accounts

### PR DESCRIPTION
**Pull request by Vassyli**
Updated Pull Request. Tests and proper authentication error message have now been implemented. User Query is something for a different PR.

### CODE REVIEW

File: src/Services/UserManagerService.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        if ($this->findByEmail($email) !== null) {
+            throw new UserEmailExistsException();
+        }
+        
+        $user = new User($name, $email, $password);
+        
+        try {
+            $user->save($entityManager);
+        } catch (DBALException $ex) {
+            throw new CrateException("An unknown DBALException occured.");
```
Comment (austenmc) [inline src/Services/UserManagerService.php]: Should probably pass on the exception messages.

**Comment by austenmc** [comment]
This lgtm and I looked over everything, but Im still not well-versed in this code honestly.

**Comment by Vassyli** [comment]
I've adjusted the PR wording to reflect the improved content of this PR.

**Comment by Vassyli** [comment]
And I'll push it to master.

**Pull request merged**